src/Icicle/Avalanche/Statement/Flatten.hs

```diff
@@ -283,6 +283,54 @@ flatX a_fresh xx stm
        -> lift $ Left $ FlattenErrorPrimBadArgs p xs
 
 
+      Core.PrimLatest (Core.PrimLatestMake t)
+       | [i] <- xs
+       -> flatX' i
+       $  \i'
+       -> do   accN      <- fresh
+               let bufT   = BufT t
+               let accT   = Mutable
+               let fpBuf  = xPrim (Flat.PrimBuf $ Flat.PrimBufMake t)
+               stm'      <- stm (xVar accN)
+
+               return
+                 $ InitAccumulator
+                     (Accumulator accN accT bufT (fpBuf `xApp` i'))
+                     (Read accN accN accT bufT stm')
```

Comment:
bit confused by why this is initialising a mutable accumulator if it's never written to. could it just be a Let?
in the future, perhaps InitAccumulator and co should be removed and replaced with reference primitives